packages/typespec/src/components/namespace/file-level.tsx

```diff
@@ -0,0 +1,52 @@
+import { Children, Declaration, Name, Namekey, Refkey, Scope, Show, useBinder } from "@alloy-js/core";
+import { useSourceFileScope } from "../../scopes/source-file.js";
+import { getGlobalNamespace } from "../../contexts/global-namespace.js";
+import { NamespaceSymbol } from "../../symbols/namespace.js";
+import { createNamespaceSymbol } from "../../symbols/factories.js";
+import { NamespaceScope } from "../../scopes/namespace.js";
+
+export interface FileLevelNamespaceProps {
```

Comment:
I think there can just be one Namespace component. What we do in C# is render block namespaces if you have a namespace within a source file, otherwise blockless namespaces when the namespace component is outside a source file. Essentially, if an entire source file is inside some namespace, we always emit the blockless namespace.